src/Utilities/Callbacks/Callbacks.jl

```diff
@@ -358,41 +372,35 @@ function CB_constructor(interval::String, solver_config, default = nothing)
     bl = dg.balance_law
     secs_per_day = day(bl.param_set)
 
-    if endswith(interval, "smonths")
-        ticks = 30.0 * secs_per_day * parse(Float64, interval[1:(end - 7)])
-        return (func) ->
-            GenericCallbacks.EveryXSimulationTime(func, ticks, solver)
-    elseif endswith(interval, "sdays")
-        ticks = secs_per_day * parse(Float64, interval[1:(end - 5)])
-        return (func) ->
-            GenericCallbacks.EveryXSimulationTime(func, ticks, solver)
-    elseif endswith(interval, "shours")
-        ticks = 60.0 * 60.0 * parse(Float64, interval[1:(end - 6)])
-        return (func) ->
-            GenericCallbacks.EveryXSimulationTime(func, ticks, solver)
-    elseif endswith(interval, "smins")
-        ticks = 60.0 * parse(Float64, interval[1:(end - 5)])
-        return (func) ->
-            GenericCallbacks.EveryXSimulationTime(func, ticks, solver)
-    elseif endswith(interval, "ssecs")
-        ticks = parse(Float64, interval[1:(end - 5)])
-        return (func) ->
-            GenericCallbacks.EveryXSimulationTime(func, ticks, solver)
-    elseif endswith(interval, "steps")
-        steps = parse(Int, interval[1:(end - 5)])
-        return (func) -> GenericCallbacks.EveryXSimulationSteps(func, steps)
-    elseif endswith(interval, "hours")
-        secs = 60 * 60 * parse(Int, interval[1:(end - 5)])
-        return (func) ->
-            GenericCallbacks.EveryXWallTimeSeconds(func, secs, mpicomm)
-    elseif endswith(interval, "mins")
-        secs = 60 * parse(Int, interval[1:(end - 4)])
-        return (func) ->
-            GenericCallbacks.EveryXWallTimeSeconds(func, secs, mpicomm)
-    elseif endswith(interval, "secs")
-        secs = parse(Int, interval[1:(end - 4)])
-        return (func) ->
-            GenericCallbacks.EveryXWallTimeSeconds(func, secs, mpicomm)
+    if (m = match(r"^([0-9\.]+)(smonths|sdays|shours|smins|ssecs)$",interval); m !== nothing)
```

Comment:
Not sure this change makes the code any clearer or more efficient, tbh. But this is fine too. Intervals will need to change to support #1085 anyway.

Reply:
Yeah, I know you weren't a fan, but it made it a bit easier to see which callback was used.